docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "The Netflix Prize dataset (`Netflix-prize`) is available on [Kaggle](https://www.kaggle.com/netflix-inc/netflix-prize-data). \n",
+    "\n",
+    "We will setup the Python [`kaggle`](https://pypi.org/project/kaggle/) official API to automatically download and use the dataset."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**(A)** Let's first setup the `DATASET_FOLDER` where the dataset will be downloaded and saved:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Run to set up Data and Output folders { display-mode: \"form\" }\n",
+    "import os\n",
+    "from pathlib import Path\n",
+    "\n",
+    "\n",
+    "NETFLIX_PRIZE_FOLDER = Path(\"netflix_prize_dataset\")\n",
+    "DATASET_FOLDER = NETFLIX_PRIZE_FOLDER / \"data\"\n",
+    "OUTPUT_FOLDER = NETFLIX_PRIZE_FOLDER  / \"outputs\"\n",
+    "\n",
+    "DATA_FILE = str(DATASET_FOLDER / \"movie_views.txt\")\n",
+    "OUTFILE_TEMPLATE = str(OUTPUT_FOLDER / '{}.txt')\n",
+    "\n",
+    "os.makedirs(DATASET_FOLDER, exist_ok=True)\n",
+    "os.makedirs(OUTPUT_FOLDER, exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**(B)** Download the `Netflix-prize` dataset from Kaggle"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Run to setup `kaggle` Python API { display-mode: \"form\" }\n",
+    "try:\n",
+    "    import kaggle\n",
+    "except ImportError:\n",
+    "    !pip install kaggle\n",
+    "    clear_output()\n",
+    "except OSError:\n",
+    "    print(\"Kaggle package is available, but credentials needs to be setup.\")\n",
+    "else:\n",
+    "    print(\"kaggle package imported correctly ✅\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Create and place your API token**\n",
+    "\n",
+    "In order to download datasets using the Kaggle API, please create and download your API token from Kaggle website:\n",
+    "\n",
+    "1. Log in to [Kaggle](https://www.kaggle.com/) website.\n",
+    "2. Go to your **Account** (locate your profile avatar in the top-right corner of the page)\n",
+    "\n",
+    "![kaggle-account](images/kaggle_account.png)\n",
+    "\n",
+    "3. Scroll down your account page until you find the **API cell**.\n",
+    "4. Click on to \"Create new API token\" button.\n",
+    "\n",
+    "![kaggle-api-key](images/api_key.png)\n",
+    "\n",
+    "This will download a file called `kaggle.json`. This file contains the credentials required to use the Kaggle APIs. \n",
+    "\n",
+    "5. Open this file, and include this data into the form below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Setup `kaggle` API credentials { display-mode: \"form\" }\n",
```

Comment:
Please combine this cell with cell "Authenticate on Kaggle via APIs" (it's better to have less cells for the user to run)

Reply:
Done